CMakeLists.txt

```diff
@@ -15,36 +17,65 @@ execute_process(
   ERROR_QUIET)
 
 if(ECODE)
-  message(
-    "Failed to determine version from git, using ${CMAKE_PROJECT_VERSION}")
+  message("Failed to determine version from git, using ${CMAKE_PROJECT_VERSION}")
   set(NEURONPARQUET_VERSION ${CMAKE_PROJECT_VERSION})
 endif()
 
+# Find required packages
 find_package(MPI REQUIRED)
 find_package(Arrow REQUIRED)
+
+# Add MPI include directories and definitions
+include_directories(SYSTEM ${MPI_INCLUDE_PATH})
+add_definitions(${MPI_CXX_COMPILE_FLAGS})
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${MPI_CXX_LINK_FLAGS}")
+add_definitions(-DOMPI_SKIP_MPICXX)
 get_filename_component(MY_SEARCH_DIR ${Arrow_CONFIG} DIRECTORY)
 find_package(Parquet REQUIRED HINTS ${MY_SEARCH_DIR})
 find_package(HDF5 REQUIRED)
 find_package(nlohmann_json REQUIRED)
 find_package(Range-v3 REQUIRED)
-
 find_package(HighFive)
+find_package(CLI11)
+find_package(Catch2)
+find_package(Python 3.8 COMPONENTS Interpreter Development.Module REQUIRED)
+
 if(NOT HighFive_FOUND)
+  set(HDF5_IS_PARALLEL ON CACHE BOOL "Enable parallel HDF5" FORCE)
   add_subdirectory(${CMAKE_SOURCE_DIR}/deps/highfive)
 endif()
 
-find_package(CLI11)
 if(NOT CLI11_FOUND)
   add_subdirectory(${CMAKE_SOURCE_DIR}/deps/cli11)
 endif()
 
-add_subdirectory(src)
-
-find_package(Catch2)
 if(NOT ${Catch2_FOUND})
   add_subdirectory(deps/catch2 EXCLUDE_FROM_ALL)
   list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/deps/catch2/extras)
 endif()
 
+execute_process(
+        COMMAND "${Python_EXECUTABLE}" -m nanobind --cmake_dir
+        OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE nanobind_ROOT)
+find_package(nanobind CONFIG REQUIRED)
```

Comment:
We normally vendor submodules for bindings, with a parameter to specify if the bindings are built.

For example here: https://github.com/BlueBrain/MorphIO/blob/master/CMakeLists.txt#L10